pkg/miscellaneous/miscellaneous.go

```diff
@@ -338,6 +338,36 @@ func ProbeEtcd(ctx context.Context, clientFactory etcdClient.Factory, logger *lo
 	return nil
 }
 
+// IsMultiNode determines whether a pod is part of a multi node setup or not
+// This is determined by checking the `initial-cluster` of the etcd configmap to check the number of members expected.
+func IsMultiNode() bool {
```

Comment:
Maybe we can align this more with that is already used here https://github.com/gardener/etcd-backup-restore/blob/a6ee3fd56bc0401fe54f048fe8166750f677a777/pkg/server/backuprestoreserver.go#L142 to not have logic duplication all over the place.

Reply:
Done